src/util/key.rs

```diff
@@ -182,6 +182,72 @@ impl PublicKey {
         buf
     }
 
+    /// Serialize the public key into a `SortKey`.
+    ///
+    /// `SortKey` is not too useful by itself, but it can be used to sort a
+    /// `[PublicKey]` slice using `sort_unstable_by_key`, `sort_by_cached_key`,
+    /// `sort_by_key`, or any of the other `*_by_key` methods on slice.
+    /// Pass the method into the sort method directly. (ie. `PublicKey::to_sort_key`)
```

Comment:
I think it'd be nice to document that the conversion is non-trivial, so cached versions may perform better.